include/static_math/trigonometry.inl

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (C) 2013-2015 Morwenn
+ *
+ * static_math is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * static_math is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program. If not,
+ * see <http://www.gnu.org/licenses/>.
+ */
+
+ /*
+ * This file is the work of theLOLflashlight's GitHub account.
+ */
+
+////////////////////////////////////////////////////////////
+// Helper functions
+
+namespace details
+{
+    constexpr size_t EXP_MAX_DEPTH = 50;
+    constexpr size_t SIN_MAX_DEPTH = 51;
+    constexpr size_t COS_MAX_DEPTH = 50;
+
+    static_assert(smath::is_odd(SIN_MAX_DEPTH), "");
+    static_assert(smath::is_even(COS_MAX_DEPTH), "");
+
+    // x^n / n!
+    template<typename T>
+    constexpr auto xn_nfac(T x, int n)
+        -> T
+    {
+        return smath::pow(x, n) / smath::factorial<T>(n);
+    }
+
+    template<typename T>
+    constexpr auto trig_helper(T x, int n, bool s)
+        -> T
+    {
+        return s ? xn_nfac(x, n) : -xn_nfac(x, n);
+    }
+
+    ////////////////////////////////////////////////////////////
+    // Function helpers
+
+    template<size_t N>
+    struct exponential
+    {
+        exponential() = delete;
+
+        static_assert(N < EXP_MAX_DEPTH, "exceeded maximum recursion depth.");
```

Comment:
I tend not to finish assertion messages with a final dot so that they look just like regular and compiler error messages. It's only a matter of style and consistency, but I really like to keep things consistent :)